### PR TITLE
[GH-1176] assemble_refbased workflow outputs schema

### PR DESCRIPTION
**Pull request by ehigham**
### Purpose

https://broadinstitute.atlassian.net/browse/GH-1176

In this PR, I'm adding a preliminary TDR schema for a Dataset containing the outputs of the `assemble_refbased` pipeline. This schema will live in testing code as:
- it serves as a placeholder for the actual schema used in production
- we can change it at any time as we learn more
- we can set-up and tear-down datasets on the fly in test code.

I have removed the use of "environments" from the datarepo module. Instead, the TDR url can be specified in an environment variable, defaulting to the dev url.
I've expanded the utilities in the `wfl.service.datarepo` namespace to accommodate creating and deleting datasets (inspired by Tom's work here https://github.com/broadinstitute/wfl/blob/709795eaee98b954a956d78f366273373c471b8c/api/src/wfl/service/datarepo.clj)

I renamed `leafname` back to `basename` and propagated this. Sorry for the noise. I also noticed that we weren't testing how the wgs module made its inputs so I've added a unit test for that,

### CODE REVIEW

File: api/src/wfl/module/sg.clj [hunk 1]
```diff
@@ -170,8 +170,7 @@
   [output pipeline {:keys [inputs uuid] :as _workflow}]
   (let [{:keys [base_file_name input_cram]} (util/parse-json inputs)
         base_file_name (or base_file_name
-                           (-> input_cram util/leafname
-                               (util/unsuffix ".cram")))
+                           (-> input_cram util/basename util/remove-extension))
```
Comment (rexwangcc) [inline api/src/wfl/module/sg.clj:173]: Like the generality here!

File: api/src/wfl/service/datarepo.clj [hunk 2]
```diff
@@ -2,84 +2,99 @@
...
+  [job-id]
+  (let [get-result #(-> (repository "jobs/" job-id "/result")
+                        (http/get {:headers (once/get-service-account-header)})
+                        util/response-body-json)
+        running?   #(-> (repository "jobs/" job-id)
+                        (http/get {:headers (once/get-service-account-header)})
+                        util/response-body-json
+                        :job_status
+                        #{"running"})]
+    (while (running?) (.sleep TimeUnit/SECONDS 1))
```
Comment (rexwangcc) [inline api/src/wfl/service/datarepo.clj:71]: Do we want to parameterize the poll interval?

File: api/src/wfl/service/datarepo.clj [hunk 3]
```diff
@@ -2,84 +2,99 @@
...
+    (get-result)))
+
+(defn create-dataset
+  "Create a dataset with EDN `dataset-request`.
+   See `DatasetRequestModel` in the DataRepo swagger page.
```
Comment (rexwangcc) [inline api/src/wfl/service/datarepo.clj:76]:
```suggestion
  "Create a dataset with EDN `dataset-request`, return the id
   of the created dataset.
   See `DatasetRequestModel` in the DataRepo swagger page.
```

Comment (rexwangcc) [reply api/src/wfl/service/datarepo.clj:76]: nit^

File: api/src/wfl/util.clj [hunk 4]
```diff
@@ -58,6 +58,10 @@ vault.client.http/http-client           ; Keep :clint eastwood quiet.
   [str]
   (json/read-str str :key-fn keyword))
 
+(defn response-body-json [response]
+  "Return the :body of the http `response` as JSON"
+  (-> response :body (or "null") parse-json))
+
```
Comment (rexwangcc) [inline api/src/wfl/util.clj:64]: nice! will update my PRs accordingly

**Review by rexwangcc — APPROVED** [review]
LGTM!

File: api/src/wfl/service/datarepo.clj (hunk 2 above)
Comment (ehigham) [reply api/src/wfl/service/datarepo.clj:71]: Not sure. We can if it causes issues in the future!

**Pull request merged**